packages/@lwc/engine-dom/src/apis/create-element.ts

```diff
@@ -105,26 +105,20 @@ export function createElement(
         );
     }
 
-    const element = document.createElement(sel);
-
-    // There is a possibility that a custom element is registered under tagName, in which case, the
-    // initialization is already carry on, and there is nothing else to do here.
-    if (!isUndefined(getAssociatedVMIfPresent(element))) {
-        return element;
-    }
-
-    const def = getComponentInternalDef(Ctor);
-    setElementProto(element, def);
-
-    createVM(element, def, {
-        tagName: sel,
-        mode: options.mode !== 'closed' ? 'open' : 'closed',
-        owner: null,
-        renderer,
+    const UpgradableConstructor = getUpgradableConstructor(sel, renderer);
+    // the custom element from the registry is expecting an upgrade callback
+    const element = new UpgradableConstructor((elm: HTMLElement) => {
+        const def = getComponentInternalDef(Ctor);
+        setElementProto(elm, def);
+
+        createVM(elm, def, {
+            tagName: sel,
+            mode: options.mode !== 'closed' ? 'open' : 'closed',
+            owner: null,
+            renderer,
+        });
+        setHiddenField(elm, ConnectingSlot, connectRootElement);
+        setHiddenField(elm, DisconnectingSlot, disconnectRootElement);
     });
-
-    setHiddenField(element, ConnectingSlot, connectRootElement);
-    setHiddenField(element, DisconnectingSlot, disconnectRootElement);
-
     return element;
```

Comment:
When invoking `createElement` from `lwc`, I think the expectation is that the returned element is an instance of the passed LWC component constructor and not a custom element from the global CE registry. There is an opportunity here to make this method more predictable by logging or throwing an error like:

```
The "x-foo" component name is already registered in the global registry. Use 
`document.createElement("x-foo")` if you want to use the component registered 
in the global registry or use a different component name for your LWC component.
```

Reply:
well, I'm ok adding this extra check, the problem is that the second time u call it with the same tag name ,there will be a component in the registry, the upgradable one, so, it is hard to differentiate between upgradable, or not upgradable registered component. It can be done by asking whether or not the upgrade callback was invoked. I wonder what is happening today? if u use createElement, then register the web component with the same name, it breaks, but if the element is registered before createElement is invoked, then the original registered class is going to be broken. I'm just worry about the throw here, which might be a non-bc change, it seems that up to this point we better let it be than throwing. thoughts?

Reply:
>  but if the element is registered before createElement is invoked, then the original registered class is going to be broken.
I don't understand why would someone would invoke `createElement` if the CE is registered.

Instead of throwing, reporting a warning should be sufficient.

Reply:
they might not know that:

a) the element name was already claimed by someone else in the APP, or
b) it was claimed by LWC in another template who uses the same name.